### PR TITLE
Add cpuinfo

### PR DESCRIPTION
**Pull request by A2va**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/c/cpuinfo/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    set_license("BSD 2-Clause")
+
+    add_urls("https://github.com/pytorch/cpuinfo.git")
+    add_versions("2022.09.15", "de2fa78ebb431db98489e78603e4f77c1f6c5c57")
+
+    add_deps("cmake")
+
+    on_load(function (package)
+        -- use main branch and not master
+        package:version_set("main", "branch")
```
Comment (waruqi) [inline packages/c/cpuinfo/xmake.lua:13]: we need not it.

File: packages/c/cpuinfo/xmake.lua [hunk 2]
```diff
@@ -11,7 +11,6 @@ package("cpuinfo")
     on_install(function (package)
         local configs = {}
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
-        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/c/cpuinfo/xmake.lua]: why?

Comment (waruqi) [reply packages/c/cpuinfo/xmake.lua]: we need it.

Comment (waruqi) [reply packages/c/cpuinfo/xmake.lua]: if it do not shared or static, you need set

https://github.com/xmake-io/xmake-repo/blob/0fd68bc72a2172fe15f5939e90b7cc4fe439cd35/packages/v/vc/xmake.lua#L10

Comment (A2va) [reply packages/c/cpuinfo/xmake.lua]: It failed on the CI and on my computer

Comment (waruqi) [reply packages/c/cpuinfo/xmake.lua]: you should fix it or add `add_configs` to disable it.

Comment (waruqi) [reply packages/c/cpuinfo/xmake.lua]: and it only for windows, but you disable them for all platforms.

**Comment by A2va** [comment]
Here a recap from the CI build.
* MingW (32 and 64): failed at `checking for c++ snippet(test)`, linker error with `clog_vlog_fatal`.
* Most of shared build doesn't compile (Windows, macOS, Linux), also failed at `checking for c++ snippet(test)`, linker error with `cpuinfo_initialize`.
* A lot of arm build also failed.

**Comment by A2va** [comment]
thanks!

**Pull request merged**